nixos/tests/taler/common/nodes.nix

```diff
@@ -0,0 +1,200 @@
+            settings = {
+              taler.CURRENCY = CURRENCY;
+            };
+            includes = [ ../conf/taler-accounts.conf ];
+            exchange = {
+              enable = true;
+              debug = true;
+              openFirewall = true;
+              denominationConfig = lib.readFile ../conf/taler-denominations.conf;
+              settings = {
+                exchange = {
+                  MASTER_PUBLIC_KEY = "2TQSTPFZBC2MC4E52NHPA050YXYG02VC3AB50QESM6JX1QJEYVQ0";
+                  BASE_URL = "http://exchange:8081/";
+                };
+                exchange-offline = {
+                  MASTER_PRIV_FILE = "${../conf/private.key}";
+                };
+              };
+            };
+          };
+        };
+      };
+
+    bank =
+      { config, ... }:
+      mkNode {
+        sshPort = 2222;
+        webuiPort = 8082;
+
+        nodeSettings = {
+          services.libeufin.bank = {
+            enable = true;
+            debug = true;
+
+            openFirewall = true;
+            createLocalDatabase = true;
+
+            initialAccounts = [
+              {
+                username = "exchange";
+                password = "exchange";
+                name = "Exchange";
+              }
+            ];
+
+            settings = {
+              libeufin-bank = {
+                WIRE_TYPE = "x-taler-bank";
+                # WIRE_TYPE = "iban";
+                X_TALER_BANK_PAYTO_HOSTNAME = "bank:8082";
+                # IBAN_PAYTO_BIC = "SANDBOXX";
+                BASE_URL = "bank:8082";
+
+                # Allow creating new accounts
+                ALLOW_REGISTRATION = "yes";
+
+                # A registration bonus makes withdrawals easier since the
+                # bank account balance is not empty
+                REGISTRATION_BONUS_ENABLED = "yes";
+                REGISTRATION_BONUS = "${CURRENCY}:100";
+
+                DEFAULT_DEBT_LIMIT = "${CURRENCY}:500";
+
+                # NOTE: The exchange's bank account must be initialised before
+                # the main bank service starts, else it doesn't work.
+                # The `services.libeufin.bank.initialAccounts` option can be used to do this.
+                ALLOW_CONVERSION = "yes";
+                ALLOW_EDIT_CASHOUT_PAYTO_URI = "yes";
+
+                SUGGESTED_WITHDRAWAL_EXCHANGE = "http://exchange:8081/";
+
+                inherit CURRENCY FIAT_CURRENCY;
+              };
+            };
+          };
+
+          services.libeufin.nexus = {
+            enable = true;
+            debug = true;
+
+            openFirewall = true;
+            createLocalDatabase = true;
+
+            settings = {
+              # https://docs.taler.net/libeufin/setup-ebics-at-postfinance.html
+              nexus-ebics = {
+                # == Mandatory ==
+                CURRENCY = FIAT_CURRENCY;
+                # Bank
+                HOST_BASE_URL = "https://isotest.postfinance.ch/ebicsweb/ebicsweb";
```

Comment:
This means the test at line 226 doesn't run in this suite -- is that the idea? Would it be possible to run a service as part of the test and point `HOST_BASE_URL` there?

Reply:
> This means the test at [line 231](https://github.com/NixOS/nixpkgs/blob/9425b7cc6895eba722c1080b35addc6992d38496/nixos/tests/taler/tests/basic.nix#L231) doesn't run in this suite, since the curl to an external URL fails -- is that the idea?

That's correct. If we can connect to the PostFinance platform, then we run the Nexus' currency conversion test, else we skip it. This was done because the NixOS test doesn't have access to the internet unless it's ran interactively, which is indicated with [a message](https://github.com/NixOS/nixpkgs/blob/9425b7cc6895eba722c1080b35addc6992d38496/nixos/tests/taler/tests/basic.nix#L7-L12) at the beginning.

> Would it be possible to run a service as part of the test and point `HOST_BASE_URL` there?

Only if we can run an EBICS/PostFinance server, which I don't think exists in nixpkgs, yet. I also don't know if any self-hostable solutions exist either or how much effort it is to package them it they do.